File: backend/app/schemas/feedback.py
from pydantic import BaseModel
from typing import Optional

class FeedbackCreateRequest(BaseModel):
    learning_item_id: Optional[str] = None
    resource_title: str
    difficulty_rating: str  # too_easy, good, okay, too_difficult
    usefulness_rating: Optional[int] = 5
    useful: Optional[bool] = True
    comment: Optional[str] = None

class FeedbackResponse(BaseModel):
    id: str
    resource_title: str
    difficulty_rating: str
    usefulness_rating: int
    useful: bool
    ai_response_text: str
    adaptation_triggered: bool
    adaptation_summary: Optional[str] = None

File: src/api/feedback.ts
import { apiClient } from './client';

export interface FeedbackData {
  resource_title: string;
  difficulty_rating: string;
  usefulness_rating?: number;
  useful?: boolean;
  comment?: string;
  learning_item_id?: string;
}

export interface FeedbackResponse {
  id: string;
  resource_title: string;
  difficulty_rating: string;
  usefulness_rating: number;
  useful: boolean;
  ai_response_text: string;
  adaptation_triggered: boolean;
  adaptation_summary?: string;
}

export const feedbackApi = {
  submitFeedback: async (data: FeedbackData): Promise<FeedbackResponse> => {
    return apiClient<FeedbackResponse>('/feedback', {
      method: 'POST',
      body: JSON.stringify(data),
    });
  },

  getFeedbacks: async (): Promise<FeedbackResponse[]> => {
    return apiClient<FeedbackResponse[]>('/feedback');
  },
};
